### PR TITLE
Turn Host kwargs into object's attributes every time

### PR DESCRIPTION
**Pull request by ogajduse**
Let's have the following host
```python
class IPAHost(Host):
    def __init__(self, sat_obj, **kwargs):
        self.satellite = sat_obj
        kwargs['hostname'] = kwargs.get('hostname', settings.ipa.hostname)
        # Allow the class to be constructed from kwargs
        kwargs['from_dict'] = True
        kwargs.update(
            {
                'base_dn': settings.ipa.basedn,
                'disabled_user_ipa': settings.ipa.disabled_ipa_user,
            }
        )
        super().__init__(**kwargs)
```

Then the following code snippet
```python
sat = Host('foo.bar')
ipahost = IPAHost(sat)
ipahost.hostname
```
will fail with `AttributeError: 'IPAHost' object has no attribute 'hostname'`

The `Host` `__init__` stores the `hostname` and the `name` in args, not in `kwargs`. So we need to set `name` and `hostname`  directly regardless of what `from_dict` is set to.

Ref: SatelliteQE/robottelo#11445
This is most likely the first host that is using `from_dict`. :smirk:

### CODE REVIEW

**Comment by ogajduse** [comment]
Another approach I can think of is to remove the `from_dict` logic and update `__dict__` every time.

**Comment by JacobCallahan** [comment]
I like your last comment. Additionally, now that hostname and name aren't technically required. We can remove them as named arguments and just take it all in as kwargs. If we go that route though, it would be good to add a docstring telling people which combination of arguments are required.

**Comment by Griffin-Sullivan** [comment]
Should I review this version or are we changing the PR? @ogajduse

**Comment by ogajduse** [comment]
@Griffin-Sullivan @JacobCallahan Implemented.
Feel free to review it now.

File: broker/hosts.py [hunk 1]
```diff
@@ -6,30 +6,45 @@
...
+        self.username = kwargs.pop("username", settings.HOST_USERNAME)
+        self.password = kwargs.pop("password", settings.HOST_PASSWORD)
+        self.timeout = kwargs.pop(
             "connection_timeout", settings.HOST_CONNECTION_TIMEOUT
         )
-        self.port = kwargs.get("port", settings.HOST_SSH_PORT)
-        self.key_filename = kwargs.get("key_filename", settings.HOST_SSH_KEY_FILENAME)
+        self.port = kwargs.pop("port", settings.HOST_SSH_PORT)
+        self.key_filename = kwargs.pop("key_filename", settings.HOST_SSH_KEY_FILENAME)
+        self.__dict__.update(kwargs) # Make every other kwarg an attribute
```
Comment (ogajduse) [inline broker/hosts.py:47]: "attribute" or "property"?

File: broker/hosts.py [hunk 2]
```diff
@@ -6,30 +6,45 @@
...
+            * *username* (``str``) --
+            Username to use for SSH connection
+            * *password* (``str``) --
+            Password to use for SSH connection
+            * *connection_timeout* (``int``) --
+            Timeout for SSH connection
+            * *port* (``int``) --
+            Port to use for SSH connection
+            * *key_filename* (``str``) --
+            Path to SSH key file to use for SSH connection
```
Comment (JacobCallahan) [inline broker/hosts.py:28]: I think this is a little bit extra 😃 
Maybe we just do something like this.
```suggestion
        Expected kwargs:
          hostname: str - Hostname or IP address of the host, required
          name: str - Name of the host
          username: str - Username to use for SSH connection
          password: str - Password to use for SSH connection
          connection_timeout: int - Timeout for SSH connection
          port: int - Port to use for SSH connection
          key_filename: str - Path to SSH key file to use for SSH connection
```

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

**Review by Griffin-Sullivan — APPROVED** [review]
Ran some functional tests and used ipython to make sure host creation works with hostname and IP and doesn't work when missing either. I also verified that changing the hostname and IP to `null` before checkin doesn't throw an error once the checkin completes.

File: broker/hosts.py [hunk 3]
```diff
@@ -6,30 +6,45 @@
...
+        """
+        logger.debug(f"Constructing host using {kwargs=}")
+        self.hostname = kwargs.get("hostname") or kwargs.get("ip", None)
```
[the remark this answers is not in the document]
Comment (ogajduse) [reply broker/hosts.py:24]: Yup, I saw that one, but I kept it there. I will remove it.

**Review by JacobCallahan — APPROVED** [review]

**Review by JacobCallahan — APPROVED** [review]

**Pull request merged**